math/common.c

```diff
@@ -99,34 +104,42 @@ double quickPow(double x, int e)
 	double res;
 	unsigned int eabs;
 
-	if (e == 0)
+	if (e == 0) {
 		return 1.0;
-	else if (e == 1)
+	}
+	else if (e == 1) {
 		return x;
+	}
 
 	res = 1.0;
 
 	if (e < 0) {
-		eabs = -e;
+		eabs = (unsigned int)(-e);
 
-		while (eabs != 0) {
-			if (eabs & 1)
+		while (eabs != 0u) {
+			if (eabs & 1u) {
 				res /= x;
+			}
 			x *= x;
 			eabs >>= 1;
 		}
 	}
 	else {
-		eabs = e;
+		eabs = (unsigned int)e;
 
-		while (eabs != 0) {
-			if (eabs & 1)
+		while (eabs != 0u) {
+			if (eabs & 1u) {
 				res *= x;
+			}
 			x *= x;
 			eabs >>= 1;
 		}
 	}
 
+	if (res > DBL_MAX) {
+		errno = ERANGE;
+	}
```

Comment:
This check is always false, double can never be greater than DBL_MAX.

Result should assume +-INF on overflow, according to the IEE754

Reply:
If `res` is `INFINITY` this check is true and I do this because on ia32 all floating point operations are done in 80-bit expanded precision registers, so checking if res is `INFINITY` using `isinf()` can be false even though after register spill its value will resolve to `INFINITY`. I agree I should add check for `-INFINITY`